docs/fee_design.md

```diff
@@ -0,0 +1,169 @@
+
+## General fee mechanism design
+
+### High level description
+
+A Child Chain operator decides and enforces the fee format and the rule for transactions. The smart contracts do not check the rules for fee transactions. The rule enforcement is perfomed by the Child chain service. This is the case as long as the Plasma network runs in POA.
+
+The transaction fee is implied for a Payment transaction. It is the difference of the sum of inputs and the sum of outputs. For instance, if the sum of inputs is 10 ETH and sum of outputs is 9.9 Eth, the 0.1 Eth difference would be the transaction fee.
+
+In order to exit fees, an operator would first put a special fee transaction into a plasma chain block. After that, the operator can spend that fee transaction output to a payment transaction. So the operator can exit it as a normal payment transaction. This removes the need to deploy an Exit Game contract for the fee transaction type as it requires only an empty contract to represent the existence of the transaction type and its protocol (using MoreVP instead of MVP).
+
+A fee transaction is special and different from a normal transaction in that:
+
+1. It does not need to consume any inputs. As the fee is implied (at least for the Payment transactions), there is no output that consumes an input.
+2. The verification of the fee transaction relies on Child Chain and watcher only. 
+
+Since the smart contracts do not verify the fee transaction, the Plasma M(ore)VP security relies on the watcher to check that the included fee transactions are following the rules correctly. If an invalid fee transaction is mined, the watcher will consider the operator as having gone rogue, and inform the users to mass exit the network.
+
+This decoupling of fee rule from smart contract gives the operator a more fine-tuned control on updating the fee rules. See the following paragraph for fee rule changes.
+
+### Fee rules upgrade/change
+
+The fee rules are not implemented in the smart contracts but by the Child Chain, which sets the fee rules, and the watcher, which informs users about the fee rules. Also, the fee transaction verification part is not bound to the logic of the fee rules. Fee rules come into force at the time the Child Chain processes transactions. A transaction not following the fee rules is rejected. Find more details on transaction verification in the design section.
+
+As a result, fee rules updates are done by changing the logic of how the child-chain service would accept/reject an incoming transaction and how watcher/wallet update the logic to follow the new fee rule when generating transaction.
```

Comment:
```suggestion
As a result, fee rules updates are done by changing the logic of how the child-chain service would accept/reject an incoming transaction and how watcher/wallet update the logic to follow the new fee rule when generating transaction.
```

child-chain or child chain or Child Chain

Reply:
Let's go with `Child Chain` and `Watcher`, will update everywhere.